extensions/wikia/ContentReview/ContentReviewImportJS.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Wikia\ContentReview;
+
+class ImportJS {
+	const IMPORT_SCRIPTS_ENTRYPOINT = 'ImportJS';
+	const IMPORT_SCRIPTS_FUNCTION = 'importWikiaScriptPages';
+	const IMPORT_SCRIPTS_KEY = 'content-review-importjs';
+	const IMPORT_SCRIPTS_VERSION = '1.0';
+
+	/**
+	 * Get js script to load imports
+	 *
+	 * @return String
+	 */
+	public function getImportScripts() {
+		$importScripts = \WikiaDataAccess::cache(
+			$this->getImportJSMemcKey(),
+			2592000, // 30 days,
+			function() {
+				$importScripts = '';
+				$title = \Title::newFromText( self::IMPORT_SCRIPTS_ENTRYPOINT, NS_MEDIAWIKI );
+
+				if ( $title instanceof \Title && $title->exists() ) {
+					$revision = \Revision::newFromTitle( $title );
+					$scripts = explode( PHP_EOL, $revision->getRawText() );
+
+					$imports = $this->prepareImports( $scripts );
+					$importScripts = $this->createInlineScript( $imports );
+				}
+
+				return $importScripts;
+			}
+		);
+
+		return $importScripts;
+	}
+
+	/**
+	 * Check and filter provided scripts.
+	 *
+	 * Names should end with '.js' extension.
+	 * External scripts are allowed only from dev.wikia.com
+	 *
+	 * @param array $scripts
+	 * @return array
+	 */
+	public function prepareImports( Array $scripts ) {
+		$imports = [];
+
+		foreach( $scripts as $key => $script ) {
+			$script = trim( $script );
+
+			if ( strtolower( substr( $script, -3 ) ) === Helper::JS_FILE_EXTENSION
+					&& !preg_match( \Title::getTitleInvalidRegex(), $script )
+			) {
+				$scriptParts = explode( ':', $script );
+				$count = count( $scriptParts );
+				// Local script
+				if ( $count === 1 ) {
+					$imports[] = $script;
+					// External script, only dev.wikia.com is allowed
+				} elseif ( $count === 2 && $scriptParts[0] === 'dev' ) {
+					$imports[] = 'external:' . $script;
+				}
+			}
+		}
+
+		return $imports;
+	}
+
+	/**
+	 * Create inline script with function to load imports
+	 *
+	 * @param array $imports
+	 * @return string
+	 */
+	public function createInlineScript( Array $imports ) {
+		$importScript = '';
+
+		if ( !empty( $imports ) ) {
+			$code = sprintf(
+				'(function(){%s(["%s"]);})();',
+				self::IMPORT_SCRIPTS_FUNCTION,
+				implode( '", "', $imports )
```

Comment:
The `escapeJsString` should be around each of the imports here, not the overall script content below.